x-pack/plugin/inference/src/test/java/org/elasticsearch/xpack/inference/services/openshiftai/action/OpenShiftAiActionCreatorTests.java

```diff
@@ -0,0 +1,761 @@
+                                -0.0123
+                            ]
+                        }
+                    ],
+                    "usage": {
+                        "prompt_tokens": 7,
+                        "total_tokens": 7,
+                        "completion_tokens": 0,
+                        "prompt_tokens_details": null
+                    }
+                }
+                """;
+            webServer.enqueue(new MockResponse().setResponseCode(413).setBody(responseJsonContentTooLarge));
+            webServer.enqueue(new MockResponse().setResponseCode(200).setBody(responseJson));
+
+            var model = createModel(getUrl(webServer), "secret", "model");
+            var actionCreator = new OpenShiftAiActionCreator(sender, createWithEmptySettings(threadPool));
+            var action = actionCreator.create(model);
+
+            PlainActionFuture<InferenceServiceResults> listener = new PlainActionFuture<>();
+            action.execute(
+                new EmbeddingsInput(List.of("abcd"), InputTypeTests.randomWithNull()),
+                InferenceAction.Request.DEFAULT_TIMEOUT,
+                listener
+            );
+
+            var result = listener.actionGet(TIMEOUT);
+
+            assertThat(result.asMap(), is(buildExpectationFloat(List.of(new float[] { 0.0123F, -0.0123F }))));
+            assertThat(webServer.requests(), hasSize(2));
+            {
+                assertNull(webServer.requests().get(0).getUri().getQuery());
+                assertThat(
+                    webServer.requests().get(0).getHeader(HttpHeaders.CONTENT_TYPE),
+                    equalTo(XContentType.JSON.mediaTypeWithoutParameters())
+                );
+                assertThat(webServer.requests().get(0).getHeader(HttpHeaders.AUTHORIZATION), equalTo("Bearer secret"));
+
+                var requestMap = entityAsMap(webServer.requests().get(0).getBody());
+                assertThat(requestMap.size(), is(2));
+                assertThat(requestMap.get("input"), is(List.of("abcd")));
+                assertThat(requestMap.get("model"), is("model"));
+            }
+            {
+                assertNull(webServer.requests().get(1).getUri().getQuery());
+                assertThat(
+                    webServer.requests().get(1).getHeader(HttpHeaders.CONTENT_TYPE),
+                    equalTo(XContentType.JSON.mediaTypeWithoutParameters())
+                );
+                assertThat(webServer.requests().get(1).getHeader(HttpHeaders.AUTHORIZATION), equalTo("Bearer secret"));
+
+                var requestMap = entityAsMap(webServer.requests().get(1).getBody());
+                assertThat(requestMap.size(), is(2));
+                assertThat(requestMap.get("input"), is(List.of("ab")));
+                assertThat(requestMap.get("model"), is("model"));
+            }
+        }
+    }
+
+    public void testExecute_ReturnsSuccessfulResponse_AfterTruncating_From400StatusCode() throws IOException {
```

Comment:
Nitpick, but the test name might be better as "testExecute_ReturnsSuccessfulResponse_AfterTruncating_From400StatusCodeWithContentTooLargeMessage", since for 400 errors without the "Please reduce your prompt; or completion length." message, we do not expect the input to be truncated. I don't think it's worth adding a test for the case where we don't truncate though, since at that point we're testing `OpenAiResponseHandler` rather than anything related to OpenShift.